tools/pythonpkg/src/include/duckdb_python/pyconnection.hpp

```diff
@@ -180,4 +177,10 @@ struct DuckDBPyConnection : public std::enable_shared_from_this<DuckDBPyConnecti
 	static void DetectEnvironment();
 };
 
+template <class T>
+static bool ModuleIsLoaded() {
```

Comment:
Does this still work properly now that we're lazily loading the Python modules?

Reply:
You have a specific test in mind?

Reply:
The idea here is that is the `pandas` module is not already loaded, an entry in the replacement scan list cannot possibly be a pandas dataframe, hence we don't need to check if it is one. Otherwise every replacement scan would involve loading all modules for which we support replacement scans (which is expanding now - pandas, polars, pyarrow, pytorch, tensorflow, ...).

The alternative would be to do a string comparison on the type instead, which we can do without loading the module.